cli/azd/cmd/env.go

```diff
@@ -403,7 +412,16 @@ func (ef *envRefreshAction) Run(ctx context.Context) (*actions.ActionResult, err
 		return nil, fmt.Errorf("initializing provisioning manager: %w", err)
 	}
 
-	getStateResult, err := ef.provisionManager.State(ctx)
+	// If resource group is defined within the project but not in the environment then
+	// add it to the environment to support BYOI lookup scenarios like ADE
+	// Infra providers do not currently have access to project configuration
```

Comment:
What do you think is the ideal behavior here? The comment here reads a little "todo" like to me, specifically: "Infra providers not having access to project configuration". We should make a decision on what's a better design:

1. Should we mirror the project configuration into environment state?
2. Or should we leave the project configuration as-is and read from there?

Reply:
Ideally we should be able to read the project configuration but as designed right now would cause a circular dependency between packages and would need to be resolved first.